src/Features/CSharp/Portable/PullMemberUp/CSharpMembersPullerService.cs

```diff
@@ -0,0 +1,86 @@
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.  
+
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CodeRefactorings.PullMemberUp
+{
+    [ExportLanguageService(typeof(IMembersPullerService), LanguageNames.CSharp), Shared]
+    internal class CSharpMembersPullerService : AbstractMembersPullerService<UsingDirectiveSyntax>
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public CSharpMembersPullerService()
+        {
+        }
+
+        protected override SyntaxTriviaList GetFirstMemberTrivia(SyntaxNode root)
+        {
+            var members = root switch
+            {
+                CompilationUnitSyntax compilationUnit => compilationUnit.Members,
+                NamespaceDeclarationSyntax namespaceDeclaration => namespaceDeclaration.Members,
+                _ => throw ExceptionUtilities.UnexpectedValue(root)
+            };
+
+            if (members.Count == 0)
+            {
+                return SyntaxTriviaList.Empty;
+            }
+
+            return members.First().GetLeadingTrivia();
+        }
+
+        protected override SyntaxNode EnsureLeadingTriviaBeforeFirstMember(SyntaxNode root, SyntaxTriviaList trivia)
+        {
+            var members = root switch
+            {
+                CompilationUnitSyntax compilationUnit => compilationUnit.Members,
+                NamespaceDeclarationSyntax namespaceDeclaration => namespaceDeclaration.Members,
+                _ => throw ExceptionUtilities.UnexpectedValue(root)
+            };
+
+            if (members.Count == 0)
+            {
+                return root;
+            }
+
+            var firstMember = members.First();
+            return root.ReplaceNode(firstMember, firstMember.WithLeadingTrivia(trivia));
+        }
+
+        protected override async Task<IEnumerable<UsingDirectiveSyntax>> GetImportsAsync(Document document, CancellationToken cancellationToken)
+        {
+            var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            return root.DescendantNodesAndSelf()
+                .Where((node) => node is CompilationUnitSyntax || node is NamespaceDeclarationSyntax)
+                .SelectMany(node => node switch
+                {
+                    CompilationUnitSyntax c => c.Usings,
+                    NamespaceDeclarationSyntax n => n.Usings,
+                    _ => default,
+                })
+                .Distinct()
```

Comment:
I might be wrong here. If I have 
```
using BarNS;
namespace Foo
{
    using BarNs;
}
```
Would the second usings be dropped? If you have verified this then just ignore this comment : )

Reply:
Ah, yes, good point. I realize now that `Distinct()` doesn't actually do anything because all statements are different in location. However, these should be dealt with either if the destination has the import (in which case we don't add the duplicate using), or when we remove unnecessary usings. 
I'll remove that distinct call, and add a test with duplicate using statements to make sure. 
For some efficiency, I could probably filter the list based on whether they are the same namespace as well, but I'm not sure how useful that would be since we should expect in most cases that there are no duplicate usings.